govcd/vapp.go

```diff
@@ -1332,3 +1332,59 @@ func (vapp *VApp) GetVMByNameOrId(identifier string, refresh bool) (*VM, error)
 	}
 	return entity.(*VM), err
 }
+
+// QueryVappList returns a list of all vApps in all the organizations available to the caller
+func (client *Client) QueryVappList() ([]*types.QueryResultVAppRecordType, error) {
+	var vappList []*types.QueryResultVAppRecordType
+	queryType := client.GetQueryType(types.QtVapp)
+	params := map[string]string{
+		"type":          queryType,
+		"filterEncoded": "true",
+	}
+	vappResult, err := client.cumulativeQuery(queryType, nil, params)
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp list : %s", err)
+	}
+	vappList = vappResult.Results.VAppRecord
+	if client.IsSysAdmin {
+		vappList = vappResult.Results.AdminVAppRecord
+	}
+	return vappList, nil
+}
+
+// getOrgInfo finds the organization to which the vApp belongs (through the VDC), and returns its name and ID
+func (vapp *VApp) getOrgInfo() (orgInfoType, error) {
```

Comment:
same remarks here if previous functions will be changed

Reply:
done